### PR TITLE
Refactor fold native MHInterpreter

### PR DESCRIPTION
**Pull request by theresa-m**
- Eliminates an extra memcpy in foldArguments MethodHandle calls
- Makes code easier to understand - reason for copying is to make stack compatible with insertPlaceHolderFrame

@DanHeidinga as discussed previously for #3980 

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

File: runtime/vm/MHInterpreter.cpp [hunk 1]
```diff
@@ -1507,48 +1507,36 @@ VM_MHInterpreter::foldForFoldArguments(j9object_t methodHandle)
...
 	U_32 combinerArgSlots = getMethodTypeArgSlots(combinerType);
 
-	/* Replace foldHandle with combinerHandle and adjust sp according to combinerHandle.type.argSlots */
-	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle PlaceHolderFrame combinerHandle args] */
-	((j9object_t *)_currentThread->sp)[foldArgSlots] = combinerHandle;
-	UDATA *spCombinerSlot = _currentThread->sp + foldArgSlots;
+	/* Add combinerHandle to top of stack and adjust sp to make room for combinerHandle.type.argSlots */
+	/* [... foldHandle args descriptionBytes MethodTypeFrame combinerHandle combinerArgs] */
+	_currentThread->sp -= (combinerArgSlots  + 1);
+	((j9object_t *)_currentThread->sp)[combinerArgSlots] = combinerHandle;
```
Comment (DanHeidinga) [inline runtime/vm/MHInterpreter.cpp]: Is this required?  If it's overwritten by the insertPlaceHolderFrame below, then I think we can skip it here

Comment (theresa-m) [reply runtime/vm/MHInterpreter.cpp]: you're right its not required. I've removed a similar line in `filterArgumentsWithCombiner` as well

File: runtime/vm/MHInterpreter.cpp [hunk 2]
```diff
@@ -1600,9 +1586,18 @@ VM_MHInterpreter::foldForFoldArguments(j9object_t methodHandle)
...
 		}
-		_currentThread->sp = spCombinerSlot;
+		/* after loop spCombinerSlot is at the top of the combiner arguments */
+		Assert_VM_true(spCombinerSlot == _currentThread->sp);
 	}
 
+	/* [... foldHandle args descriptionBytes MethodTypeFrame foldHandle PlaceHolderFrame foldHandle combinerArgs] */
+	insertPlaceHolderFrame(combinerArgSlots, methodHandle, J9VMJAVALANGINVOKEMETHODHANDLE_FOLDHANDLEPLACEHOLDER_METHOD(_vm));
+
+	/* insertPlaceHolderFrame will replace combinerHandle with methodHandle. This should be undone so the stack is correctly
```
Comment (DanHeidinga) [inline runtime/vm/MHInterpreter.cpp]: This comment isn't correct any more as the MH isn't written until the end.  Can you update it here and in the other case as well?

Comment (theresa-m) [reply runtime/vm/MHInterpreter.cpp]: updated.

**Comment by DanHeidinga** [comment]
Jenkins test sanity aix,win,xlinux jdk11

**Pull request merged**